carma_wm/include/carma_wm/SignalizedIntersectionManager.hpp

```diff
@@ -144,6 +155,69 @@ class SignalizedIntersectionManager
   */
   lanelet::Lanelets identifyInteriorLanelets(const lanelet::Lanelets& entry_llts, const std::shared_ptr<lanelet::LaneletMap>& map);
 
+  /**
+   * @brief processSpatFromMsg update map's traffic light states with SPAT msg
+   *
+   * @param spat_msg Msg to update with
+   */
+  void processSpatFromMsg(const carma_v2x_msgs::msg::SPAT& spat_msg,
+    const std::shared_ptr<lanelet::LaneletMap>& semantic_map);
+
+  /*!
+  *  \brief Update the recorded traffic signal objects in the intersection as fixed signal,
+  *         determining the phase cycles from the available signal durations.
+  *         Do nothing if intersection_id is not found.
+  *  \param intersection_id of the intersection
+  *  \param lanelet_map to update the regulatory elements according to internal signals recorded
+  */
+  void updateSignalAsFixedSignal(uint16_t intersection_id,
+    const std::shared_ptr<lanelet::LaneletMap>& semantic_map);
+
+  /*!
+  *  \brief Update the recorded traffic signal objects in the intersection as dynamic signal,
+  *         directly overwriting any previous states
+  *  \param intersection_id of the intersection
+  *  \param lanelet_map to update the regulatory elements according to internal signals recorded
+  */
+  void updateSignalAsDynamicSignal(uint16_t intersection_id,
+    const std::shared_ptr<lanelet::LaneletMap>& semantic_map);
+
+  /*!
+  *  \brief Extract tuple of 1) vector of pair of traffic signal states' min_end_time and their
+            states, which is mainly used by CARMA to comply with traffic signals.
+            2) vector of traffic signal states' start_times that match in order
+  *  \param curr_intersection carma_v2x_msgs::msg::IntersectionState from SPAT
+  *  \param current_movement_state carma_v2x_msgs::msg::MovementState from SPAT
+  *  \return tuple of 2 vectors. 1) vector of min_end_time and state pairs 2) vector of start_time
+  */
+  std::tuple<std::vector<std::pair<boost::posix_time::ptime, lanelet::CarmaTrafficSignalState>>,
+    std::vector<boost::posix_time::ptime>>
+    extract_signal_states_from_movement_state(
+      const carma_v2x_msgs::msg::IntersectionState& curr_intersection,
+      const carma_v2x_msgs::msg::MovementState& current_movement_state);
+
+    /*! \brief update minimum end time to account for minute of the year
+    * \param min_end_time minimum end time of the spat movement event list
+    * \param moy_exists tells weather minute of the year exist or not
+    * \param moy value of the minute of the year
+    * \param use_real_time_spat_in_sim Boolean to indicate if the incoming spat is based on
+    *   wall clock. Required in edge cases where deployment in simulation is receiving
+    *   SPaT messages based on wall clock.
+    * Defaults to false.
```

Comment:
nit: Can we move this defaults to false to the line above - it seems like a statement for the entire function here

Reply:
Removed.